samcli/commands/init/init_templates.py

```diff
@@ -59,7 +63,7 @@ def prompt_for_location(self, runtime, dependency_manager):
             return (template_md["init_location"], "hello-world")
         if template_md.get("directory") is not None:
             return (os.path.join(self.repo_path, template_md["directory"]), template_md["appTemplate"])
-        raise UserException("Invalid template. This should not be possible, please raise an issue.")
+        raise InvalidInitTemplateError("Invalid template. This should not be possible, please raise an issue.")
```

Comment:
For template issue we used name `TemplateNotFoundException`, but here we are calling it an `Error`. How are we differentiating between `Error` and `Exception` terms?

Reply:
An Error typically indicates serious problem that an application should not try to catch, while an Exception indicates conditions that an application might want to catch.

We can't really do anything about an Invalid template here so decided to go with Error instead of Exception.